Src/AmrCore/AMReX_AmrParticles.H

```diff
@@ -249,40 +249,47 @@ ParticleToMesh (PC const& pc, const Vector<MultiFab*>& mf,
     }
 }
 
-template <int NStructReal, int NStructInt=0, int NArrayReal=0, int NArrayInt=0,
+template <typename T_ParticleType, int NArrayReal=0, int NArrayInt=0,
           template<class> class Allocator=DefaultAllocator>
-class AmrParticleContainer // NOLINT
-    : public ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
+class AmrParticleContainer_impl
+    : public ParticleContainer_impl<T_ParticleType, NArrayReal, NArrayInt, Allocator>
 {
 
 public:
 
-    typedef Particle<NStructReal, NStructInt> ParticleType;
+    using ParticleType = T_ParticleType;
 
-    AmrParticleContainer () = default;
+    AmrParticleContainer_impl ()
+        : ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator>()
+    {
+    }
 
-    AmrParticleContainer (AmrCore* amr_core)
-        : ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>(amr_core->GetParGDB())
+    AmrParticleContainer_impl (AmrCore* amr_core)
+        : ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator>(amr_core->GetParGDB())
     {
     }
 
-    AmrParticleContainer (const Vector<Geometry>            & geom,
-                          const Vector<DistributionMapping> & dmap,
-                          const Vector<BoxArray>            & ba,
-                          const Vector<int>                 & rr)
-        : ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>(geom, dmap, ba, rr)
+    AmrParticleContainer_impl (const Vector<Geometry>            & geom,
+                               const Vector<DistributionMapping> & dmap,
+                               const Vector<BoxArray>            & ba,
+                               const Vector<int>                 & rr)
+        : ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator>(geom, dmap, ba, rr)
     {
     }
 
-    ~AmrParticleContainer () override = default;
+    ~AmrParticleContainer_impl () override = default;
```

Comment:
Looks like clang tidy wants us to write
```suggestion
    virtual ~AmrParticleContainer_impl () = default;
```
or maybe
```suggestion
    virtual ~AmrParticleContainer_impl () override = default;
```

Reply:
Much discussion on `virtual` vs. `override` as specifiers for destructors of derived classes online...

Reply:
update: haha
```
error: prefer using 'override' or (rarely) 'final' instead of 'virtual' [modernize-use-override,-warnings-as-errors]
    virtual ~AmrParticleContainer_impl () = default;
    ~~~~~~~~^
                                          override 
Suppressed 41424 warnings (41389 in non-user code, 35 NOLINT).
```

Reply:
I think there is a bug in clang-tidy regarding virtual class destructor. You can use nolint to get rid of it. For example,
```
$ git grep -i "nolint" Src/Particle/
Src/Particle/AMReX_NeighborParticles.H:class NeighborParticleContainer // NOLINT(cppcoreguidelines-virtual-class-destructor) // clang-tidy seems wrong.
```

Reply:
Thanks! Will do

Reply:
I see we actually removed a `// NOLINT` for this class by accident in the diff.